plugins/modules/host.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2024 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+)
+from cm_client import ApiHost,ApiHostList
+from cm_client import ClustersResourceApi
+from cm_client import HostsResourceApi
+from cm_client.rest import ApiException
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: host
+short_description: Manage hosts within Cloudera Manager
+description:
+  - Allows for the management of hosts within the Cloudera Manager.
+  - It provides functionalities to create, delete, attach, or detach host instance from a cluster.
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+options:
+  host_name:
```

Comment:
Similar to the common `version` parameter, this (if normalized to `hostname`) will conflict with the common parameter.  As a temporary fix, maybe name this to be `cluster_hostname`?